api/cpp/esp-idf/slint/include/slint-esp.h

```diff
@@ -7,10 +7,57 @@
 #include "esp_lcd_touch.h"
 #include "esp_lcd_types.h"
 
+/**
+ * This data structure configures the Slint platform for use with ESP-IDF, in particular
+ * the esp_lcd component (
+ * https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/peripherals/lcd.html )
+ * for touch input and on-screen rendering.
+ *
+ * Slint supports three different ways of rendering:
+ *
+ * * Single-buffering: Allocate one frame-buffer at a location of your choosing in RAM, and
+ *                     set the `buffer1` field.
+ * * Double-buffering: Call `esp_lcd_rgb_panel_get_frame_buffer` to obtain two frame buffers
+ *                     allocated by the `esp_lcd` driver and set `buffer1` and `buffer2`.
+ * * Line-by-line rendering: Set neither `buffer1` nor `buffer2` to instruct Slint to allocate
+ *                           a buffer (with MALLOC_CAP_INTERNAL) big enough to hold one line,
+ *                           render into it, and send it to the display.
```

Comment:
We need to remove the experimental gating of the line-by-line though.

Also we need to support byte swap there too.